modules/sdk-coin-sol/src/lib/stakingRawMsgAuthorizeBuilder.ts

```diff
@@ -0,0 +1,129 @@
+} from '@bitgo/sdk-core';
+import { Transaction } from './transaction';
+import {
+  Transaction as SOLTransaction,
+  Message as SOLMessage,
+  SystemProgram,
+  SystemInstruction,
+  StakeProgram,
+} from '@solana/web3.js';
+
+import assert from 'assert';
+import BigNumber from 'bignumber.js';
+
+export class StakingRawMsgAuthorizeBuilder extends BaseTransactionBuilder {
+  protected _transaction: Transaction;
+  protected _transactionMessage: string;
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._transaction = new Transaction(_coinConfig);
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.StakingAuthorizeRaw;
+  }
+
+  /** @inheritdoc */
+  initBuilder(tx: Transaction): void {
+    if (this.validateTransaction(tx)) {
+      this.transactionMessage(tx.solTransaction.serializeMessage().toString('base64'));
+    }
+  }
+
+  /**
+   * The raw message generated by Solana CLI.
+   *
+   * @param {string} msg msg generated by 'solana stake-authorize-check.
+   * @returns {StakeBuilder} This staking builder.
+   *
+   */
+  transactionMessage(msg: string): this {
+    this._transactionMessage = msg;
+    return this;
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    assert(this._transactionMessage, 'missing transaction message');
+
+    this.transaction.solTransaction = SOLTransaction.populate(
+      SOLMessage.from(Buffer.from(this._transactionMessage, 'base64')),
+      []
+    );
+    this.transaction.setTransactionType(this.transactionType);
+    return this.transaction;
+  }
+
+  validateTransaction(tx: Transaction): boolean {
+    return this.validateMessage(tx.solTransaction.serializeMessage().toString('base64'));
+  }
+
+  async build(): Promise<BaseTransaction> {
+    this.validateMessage(this._transactionMessage);
+    return this.buildImplementation();
+  }
+
+  protected validateMessage(msg: string): boolean {
+    const tx = SOLTransaction.populate(SOLMessage.from(Buffer.from(msg, 'base64')), []);
+    const instructions = tx.instructions;
+    assert(instructions.length === 2, 'Invalid transaction, expected 2 instruction, got ' + instructions.length);
+    for (const instruction of instructions) {
+      switch (instruction.programId.toString()) {
+        case SystemProgram.programId.toString():
+          const instructionName = SystemInstruction.decodeInstructionType(instruction);
+          assert(instructionName === 'AdvanceNonceAccount', 'Invalid transaction:' + instructionName);
+          break;
+        case StakeProgram.programId.toString():
+          const data = instruction.data.toString('hex');
+          assert(data === '0a00000001000000', 'Invalid staking instruction data:' + data);
+          break;
+        default:
+          assert(
+            false,
+            'Invalid transaction, instruction program id not supported: ' + instruction.programId.toString()
+          );
```

Comment:
You can just throw an error, asserting a constant false value is not very intuitive. 

```suggestion
	throw new Error(`Invalid transaction, instruction program id not supported: ${instruction.programId.toString()}`);
```